tests/SbrpTests/Config.cs

```diff
@@ -0,0 +1,30 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+
+namespace SbrpTests;
+
+internal static class Config
+{
+    public const string RepoRootEnv = "SBRP_TESTS_REPO_ROOT";
+    public const string BuildTypeEnv = "SBRP_TESTS_BUILD_TYPE";
+
+    public static string RepoRoot
+    {
+        get
+        {
+            string repoRoot = Environment.GetEnvironmentVariable(RepoRootEnv) ??
+                Environment.CurrentDirectory.Substring(0, Environment.CurrentDirectory.IndexOf("artifacts"));
```

Comment:
Consider removing this hardcoded path expectation and the environment variable in favor of runtimeconfiguration.json values. This path is known at build time but not at run time so a runtimeconfiguration value seems most appropriate.

Reply:
Same for the BuildType which is also known at build time.

Reply:
All these paths are available at build time via msbuild properties and really shouldn't be hardcoded. It doesn't matter how small or large this repository is, we should never hardcode these paths.

- `Environment.CurrentDirectory.Substring(0, Environment.CurrentDirectory.IndexOf("artifacts"))` ->`$(RepoRoot)`
- `Path.Combine(Config.RepoRoot, "artifacts/source-build/self/src/artifacts/packages", Config.BuildType, "Shipping")` -> `$(ArtifactsShippingPackagesDir)`
- `BuildType` -> `$(Configuration)`

Reply:
This nice thing about @ViktorHofer's suggestion is the runtime config properties can be set via msbuild logic in the csproj and read at runtime.  It would eliminate any hard coding of the default values to ensure correctness.  This feels like a better option given all these settings are known at build time.

Reply:
I think the issue comes from having to work with the source-build inner clone in this case. 

Because the tests are run with "./test.sh", all of the MSBuild properties are set for the outer, non source-build repo. This is fine in the case of `RepoRoot` for the generate tests, but it gets more complicated when having to get the SB properties for the validation tests. For example, `ArtifactsShippingPackagesDir = /root/repos/source-build-reference-packages/artifacts/packages/Release/Shipping/`. I need it to be equal to `/root/repos/source-build-reference-packages/artifacts/source-build/self/src/artifacts/packages/Release/Shipping/`. This means that I'd still have to do some hardcoding to get the correct path.

I'm fine with using runtimeconfiguration.json values, but it would still involve doing some hardcoding.

Reply:
> Because the tests are run with "./test.sh", all of the MSBuild properties are set for the outer, non source-build repo.

Can we change that so that the test invocation runs in the expected context (in this case, the source-build inner clone)?

Reply:
That gets a little tricky. Since the source-build build is run using arcade infra, I believe that arcade might have to support running tests as part of the inner clone build. At the same time, I don't know if there's a way to separate building the source-build inner clone from working in the context of the source-build inner clone. That is, I think you'd have to build the inner clone in order to work with it. We'd have to essentially change the infra of this test to run as a task that is triggered by a build arg such as `--validate` and try to run that task in the `InnerSourceBuildRepoRoot`